### PR TITLE
Refine bug reporting/feature requests (fixes #236)

### LINKED ISSUE #236: Refine bug reporting / feature requests
Ask for the issue title and type (bug/feature request).

### PR DESCRIPTION
**Pull request by Trigary**
Change summary:

- The bug reporting feature has been moved from `HomeController` to its own class
- The feature got its own language resource files (instead of using `general.php`)
- Bug reporting has been generalized to "issue reporting", which includes bug reports and feature requests
  - Translations have been modified to reflect this change
  - A selector input has been added to the page, allowing users to select whether they are reporting a bug or a feature request
- Input validation has been added to the backend
- Missing GitHub auth tokens are now logged, allowing easy testing

Additional notes:

- The created issues only get the selected labels applied if the authorization used to make the request grants push access. Currently, that is not the case: no labels get applied due to a lack of permissions. This isn't a big deal: everything else still functions normally.

### CODE REVIEW

File: app/Http/Controllers/ReportIssueController.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+     */
+    public function report(Request $request): View|RedirectResponse
+    {
+        //validate input
+        $validator = Validator::make($request->all(), [
+            'title' => 'required|string',
+            'type' => 'required|in:issue.type_bug,issue.type_feature',
+            'description' => 'required|string',
+        ]);
+        $validator->validate();
```
Comment (kdmnk) [inline app/Http/Controllers/ReportIssueController.php]: no comment needed, use $request->validate([...]).

File: app/Http/Controllers/ReportIssueController.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+        //request details, removing slashes and sanitize content
+        $title = __($request->input('type'), [], 'en') . ': ' . $request->input('title');
+        $title = htmlspecialchars(stripslashes($title), ENT_QUOTES);
+        $username = htmlspecialchars(stripslashes(user()->name), ENT_QUOTES);
+        $body = htmlspecialchars(stripslashes($request->input('description')), ENT_QUOTES);
```
Comment (kdmnk) [inline app/Http/Controllers/ReportIssueController.php]: make a helper function for sanitizing

File: app/Http/Controllers/ReportIssueController.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+/**
+ * Software bug reporting.
+ */
+class ReportIssueController extends Controller
+{
+    /**
+     * Show the bug reporting page.
+     */
+    public function index(): View
```
Comment (kdmnk) [inline app/Http/Controllers/ReportIssueController.php]: call this create

File: app/Http/Controllers/ReportIssueController.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        return view('report_issue');
+    }
+
+    /**
+     * Create a new bug report or feature request. A GitHub issue is automatically created from the provided data.
+     * The returned view contains the URL of the created issue.
+     * @throws ValidationException
+     * @throws AuthenticationException
+     */
+    public function report(Request $request): View|RedirectResponse
```
Comment (kdmnk) [inline app/Http/Controllers/ReportIssueController.php]: call this store

File: routes/web.php [hunk 5]
```diff
@@ -85,8 +86,9 @@
     Route::post('/home/edit', [HomeController::class, 'editNews'])->name('home.edit');
     Route::post('/color/{mode}', [HomeController::class, 'colorMode'])->name('set-color-mode');
 
-    Route::post('/report_bug', [HomeController::class, 'reportBug'])->name('reportbug');
-    Route::get('/report_bug', [HomeController::class, 'indexReportBug'])->name('index_reportbug');
+    /** Issue reporting */
+    Route::get('/report_issue', [ReportIssueController::class, 'index'])->name('report_issue.index');
+    Route::post('/report_issue', [ReportIssueController::class, 'report'])->name('report_issue.report');
```
Comment (kdmnk) [inline routes/web.php]: call `issues/create` and `issues` .

**Review by kdmnk — CHANGES_REQUESTED** [review]
I suggetsed some smaller changes, mainly to follow this convention: https://laravel.com/docs/11.x/controllers#actions-handled-by-resource-controllers

File: app/Http/Controllers/ReportIssueController.php (hunk 3 above)
Comment (Trigary) [reply app/Http/Controllers/ReportIssueController.php]: Are you sure? Based on the document you linked, I think it should be called `index`.

**Comment by Trigary** [comment]
Thanks, I made the requested changes (except one, see the conversation).

File: app/Http/Controllers/ReportIssueController.php (hunk 3 above)
Comment (kdmnk) [reply app/Http/Controllers/ReportIssueController.php]: "index" is used to display a list of a resource. "create" is for the create page which sends the "store" request on submission. 
In another sense, index can stay as the ~main page of a feature. 
I'll let you decide.

Comment (Trigary) [reply app/Http/Controllers/ReportIssueController.php]: I see, thanks. I renamed the method.

**Review by kdmnk — APPROVED** [review]

**Pull request merged**